lisp/transient.el

```diff
@@ -512,6 +512,16 @@ slot is non-nil."
    (argument-regexp  :initarg :argument-regexp))
   "Class used for sets of mutually exclusive command-line switches.")
 
+(cl-defmethod initialize-instance ((obj transient-switches) slots)
+  (cl-call-next-method)
+  (-let [(&plist :argument-regexp :choices :argument-format) slots]
```

Comment:
I don't use `-let`. Please use something else.

Reply:
ok for `cl-destructuring-bind`?

Reply:
Use `pcase` if possible. But because I do not use `-let` I don't actually know what this does and so I also do not know if `pcase` is capable of doing the same.

Reply:
`pcase` is capable of plist destructuring through the `map` pattern, but only since Emacs 27.

Reply:
You can use `macroexpand-all` to see what `-let` have done. I use `-let` to extract `valA` `valB` from plist `'(:keyA valA :keyB valB)` with `(-let [(&plist :keyA :keyB) '(:keyA valA :keyB valB)])`.

I'm afraid that `pcase` doesn't have a convenient way to do plist destructuring. If you don't like `cl-destructuring-bind`, I will use `plist-get` directly.

BTW, It's possible to use `&key` pattern in arglist to destruct the list directly.

```
(cl-defun testfn (_ (&key a b c)) (list a b c))
(testfn 'ignored '(:a 1 :b 2 :c 3)) ;; => '(1 2 3)
```

Reply:
> I'm afraid that `pcase` doesn't have a convenient way to do plist destructuring. If you don't like `cl-destructuring-bind`, I will use `plist-get` directly.

In that case you can use `cl-destructuring-bind`.